api/machinemanager/machinemanager.go

```diff
@@ -49,6 +49,48 @@ func (client *Client) ForceDestroyMachines(machines ...string) ([]params.Destroy
 	return client.destroyMachines("ForceDestroyMachine", machines)
 }
 
+// DestroyMachinesWithParams removes the given set of machines, the semantics of which
+// is determined by the force and keep parameters.
+// TODO(wallyworld) - for Juju 3.0, this should be the preferred api to use.
+func (client *Client) DestroyMachinesWithParams(force, keep bool, machines ...string) ([]params.DestroyMachineResult, error) {
+	if !keep {
```

Comment:
Why bother keeping this block? Since we need to handle keep and !keep in the call below, this block seems pointless.